src/components/Avatar/AvatarContent.tsx

```diff
@@ -1,30 +1,46 @@
 import React from "react";
 import cx from "classnames";
-import PropTypes from "prop-types";
 import { BEMClass } from "../../helpers/bem-helper";
-import Icon from "../Icon/Icon";
-import { AVATAR_SIZES, AVATAR_TYPES } from "./AvatarConstants";
+import { AvatarSizes, AvatarTypes } from "./AvatarConstants";
 import "./AvatarContent.scss";
+import VibeComponentProps from "src/types/VibeComponentProps";
+import Icon, { iconSubComponentProps } from "../Icon/Icon";
 
 const AVATAR_CONTENT_CSS_BASE_CLASS = "monday-style-avatar-content";
 const bemHelper = BEMClass(AVATAR_CONTENT_CSS_BASE_CLASS);
-export const AvatarContent = ({ type, src, icon, text, ariaLabel, role, size, textClassName }) => {
+
+interface AvatarContentProps extends VibeComponentProps {
+  src?: string;
+  type?: AvatarTypes;
+  size?: AvatarSizes;
+  role?: string;
+  ariaLabel?: string;
+  /** we support two types of icons - SVG and FONT (classname) so this prop is either the name of the icon or the component */
+  icon?: string | React.FunctionComponent<iconSubComponentProps> | null;
+  textClassName?: string;
+  text?: string;
+}
+
+export const AvatarContent: React.FC<AvatarContentProps> & {
+  sizes?: typeof AvatarSizes;
+  types?: typeof AvatarTypes;
+} = ({ type, src, icon, text, ariaLabel, role, size, textClassName }) => {
   const className = cx(bemHelper({ element: type }), bemHelper({ element: type, state: size }));
   switch (type) {
-    case AVATAR_TYPES.IMG:
+    case AvatarTypes.IMG:
       return <img role={role} alt={ariaLabel} src={src} className={className} />;
-    case AVATAR_TYPES.ICON:
+    case AvatarTypes.ICON:
       return (
         <Icon
           icon={icon}
           aria-label={ariaLabel}
-          role={role}
+          // role={role}
```

Comment:
why ?

Reply:
The Icon component doesn't have a role prop in it